src/RequestGroupMan.cc

```diff
@@ -474,11 +484,43 @@ createInitialCommand(const std::shared_ptr<RequestGroup>& requestGroup,
 void RequestGroupMan::fillRequestGroupFromReserver(DownloadEngine* e)
 {
   removeStoppedGroup(e);
-  if (static_cast<size_t>(maxSimultaneousDownloads_) <= numActive_) {
+
+  int maxSimultaneousDownloads=maxSimultaneousDownloads_;
+
+  if(optimizeSimultaneousDownloads_) {
+    // optimize method is requested, so gauge current speed
+    int currentSpeed=getNetStat().calculateDownloadSpeed();
+    if(currentSpeed>0) {
+      // apply the rule
+      int speed=currentSpeed;
+      if((maxOverallDownloadSpeedLimit_>0) && (speed>maxOverallDownloadSpeedLimit_)) {
+        speed=maxOverallDownloadSpeedLimit_;
+      }
+      maxSimultaneousDownloads=ceil(
+        optimizeSimultaneousDownloadsCoeffA_
+        +optimizeSimultaneousDownloadsCoeffB_ * log10(speed*8./1000000.)
+      );
+      if((maxSimultaneousDownloads_>0)&&(maxSimultaneousDownloads>maxSimultaneousDownloads_)) {
```

Comment:
you are right. in fact i remember this was temporary... what i intended initially was to check whether --max-concurrent-downloads had been explicitly defined, but i did not find the right way to do it and left this in. 
The idea would be that if max-concurrent-downloads is not explicitly set, then there is no absolute maximum for the optimization. I would find it useful, please tell me your opinion and i would  complete. I found later on that every option implements a defined() method and i guess it returns false if not explicitly defined.